### ISSUE TITLE
Implement Evaluation class responsible for model metric computation

### ISSUE THREAD

**Issue by QbaTa** [Artificial-Intelligence-in-Medicine-AGH/cnn-classifier#27]
## Description
Implement a dedicated module `evaluation.py` responsible for calculating and reporting key performance metrics for the trained CNN model. This module will provide standardized functions for evaluating model predictions.

## Goals
Define a `Evaluation` class with:
- `accuracy()` - compute overall accuracy. #28 
- `precision_recall()` - compute precision & recall. #29 
- `f1_score()` - compute F1-score.
- `roc_auc()` - compute ROC-AUC.

## Implementation Notes
- File `evaluation.py`
- Class `Evaluation`
- Use `sklearn.metrics`

> [!NOTE]
> If you want to deepen your understanding, you may try implementing this metric manually (e.g. `accuracy_manual`, `precision_manual`, `recall_manual`, ...). After that, compare the results of your own implementation with the values returned by the `sklearn metrics`.
>
>This is not a required part of the task it’s purely for learning purposes.
> If you were planning to paste all the code into ChatGPT just to get it done quickly, feel free to skip this step.

## Expected Usage
```python
from evaluation import Evaluation

ev = Evaluation()
metrics = {
    "accuracy": ev.accuracy(y_true, y_pred),
    "precision_recall": ev.precision_recall(y_true, y_pred),
    "f1": ev.f1_score(y_true, y_pred),
    "roc_auc": ev.roc_auc(y_true, y_prob)
}
print(metrics)
```
